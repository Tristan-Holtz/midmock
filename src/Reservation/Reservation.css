.reservation-card {
  background-color: #fff;
  border: 3px solid turquoise;
  border-radius: 0.5em;
  height: 12em;
  width: 12em;
}

.reservation-section {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 1em;
  width: 100%;
}

.cancel-button {
  background-color: turquoise;
  height: 2.5em;
  width: 90%;
}
