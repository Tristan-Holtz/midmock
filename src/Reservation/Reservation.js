import React, { Component } from 'react';
import './Reservation.css';

const Reservation = ({ reservationList }) => {
  const allReservations = reservationList.map(reservation => {
    return (
      <section key={reservation.id} className="reservation-card">
        <h3>{reservation.name}</h3>
        <p>{reservation.date}</p>
        <p>{reservation.time}</p>
        <p>Number of guests: {reservation.number}</p>
        <button className="cancel-button">Cancel</button>
      </section>
    );
  });
  return <section className="reservation-section">{allReservations}</section>;
};

export default Reservation;
