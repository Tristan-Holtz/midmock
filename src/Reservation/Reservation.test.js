import React from 'react';
import { shallow } from 'enzyme';
import Reservation from './Reservation';

describe('Reservation', () => {
  it('should render the component', () => {
    const wrapper = shallow(
      <Reservation
        reservationList={[
          {
            id: Date.now(),
            name: '',
            date: '',
            time: '',
            number: ''
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
