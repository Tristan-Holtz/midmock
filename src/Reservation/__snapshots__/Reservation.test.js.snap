// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation should render the component 1`] = `
<section
  className="reservation-section"
>
  <section
    className="reservation-card"
  >
    <h3 />
    <p />
    <p />
    <p />
    <button
      className="cancel-button"
    >
      Cancel
    </button>
  </section>
</section>
`;
